app/frontend/src/cljs/teet/file/file_view.cljs

```diff
@@ -116,19 +119,41 @@
     (let [[sort-fn comparator] (sorters value)]
       (sort-by sort-fn comparator files))))
 
+(defn get-document-groups
+  "Returns the document groups for the file list"
+  [files]
+  (let [outs (mapv (fn [x]
+                     (-> x
+                         (assoc :file/group-code (get-in x [:file/document-group :filename/code]))
+                         (assoc :file/group-name (get-in x [:file/document-group :db/ident]))
+                         (dissoc :file/document-group))) files)]
+    outs))
+
+(defn sort-document-groups
+  "Sort document group vector by 'ungrouped' first and the rest alphabetically"
+  [doc-group first-el]
+  (mapv (fn [x] (:key x))
+        (sort-by
+          (juxt #(not= first-el (:key %)) :name)
```

Comment:
This is because the ungrouped item needs to be different in two locations. I am using the same function to make sure that :ungrouped-values is first in one map and :document.group/ungrouped is in another.